Core/Src/nero_state_machine.c

```diff
@@ -0,0 +1,67 @@
+#include "nero_state_machine.h"
+#include "stdint.h"
+#include "stdbool.h"
+#include "can_handler.h"
+#include "state_machine.h"
+
+static nero_state_t nero_state;
```

Comment:
I think this is an okay way to do it. Technically doesn't allow for more than 1 "nero" to be active in the system but like at this point I don't care about modular design lol

Reply:
Actually jk I lied I want to have this be a struct that the user manages rather than have a global reference. Imo its a more modular design that can scale better